File: z-clients/java/src/main/java/com/svix/diom/apis/AdminAuthToken.java
// this file is @generated
package com.svix.diom.apis;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.svix.diom.DiomException;
import com.svix.diom.HttpClient;
import java.util.Collections;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;
import com.svix.diom.models.AdminAuthTokenCreateIn;
import com.svix.diom.models.AdminAuthTokenCreateOut;
import com.svix.diom.models.AdminAuthTokenDeleteIn;
import com.svix.diom.models.AdminAuthTokenDeleteOut;
import com.svix.diom.models.AdminAuthTokenExpireIn;
import com.svix.diom.models.AdminAuthTokenExpireOut;
import com.svix.diom.models.AdminAuthTokenListIn;
import com.svix.diom.models.AdminAuthTokenRotateIn;
import com.svix.diom.models.AdminAuthTokenRotateOut;
import com.svix.diom.models.AdminAuthTokenUpdateIn;
import com.svix.diom.models.AdminAuthTokenUpdateOut;
import com.svix.diom.models.AdminAuthTokenWhoamiIn;
import com.svix.diom.models.AdminAuthTokenWhoamiOut;
import com.svix.diom.models.ListResponseAdminAuthTokenOut;

public class AdminAuthToken {
    private final HttpClient client;

    public AdminAuthToken(HttpClient client) {
        this.client = client;
    }

    /** Create an auth token */
    public AdminAuthTokenCreateOut create(
        final AdminAuthTokenCreateIn adminAuthTokenCreateIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.admin.auth-token.create",
            null,
            adminAuthTokenCreateIn,
            AdminAuthTokenCreateOut.class
        );
    }

    /** Expire an auth token */
    public AdminAuthTokenExpireOut expire(
        final AdminAuthTokenExpireIn adminAuthTokenExpireIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.admin.auth-token.expire",
            null,
            adminAuthTokenExpireIn,
            AdminAuthTokenExpireOut.class
        );
    }

    /** Rotate an auth token, invalidating the old one and issuing a new secret */
    public AdminAuthTokenRotateOut rotate(
        final AdminAuthTokenRotateIn adminAuthTokenRotateIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.admin.auth-token.rotate",
            null,
            adminAuthTokenRotateIn,
            AdminAuthTokenRotateOut.class
        );
    }

    /** Delete an auth token */
    public AdminAuthTokenDeleteOut delete(
        final AdminAuthTokenDeleteIn adminAuthTokenDeleteIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.admin.auth-token.delete",
            null,
            adminAuthTokenDeleteIn,
            AdminAuthTokenDeleteOut.class
        );
    }

    /** List auth tokens for a given owner */
    public ListResponseAdminAuthTokenOut list(
        final AdminAuthTokenListIn adminAuthTokenListIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.admin.auth-token.list",
            null,
            adminAuthTokenListIn,
            ListResponseAdminAuthTokenOut.class
        );
    }

    /** List auth tokens for a given owner */
    public ListResponseAdminAuthTokenOut list(
        
    ) throws DiomException {
        return this.list(
            new AdminAuthTokenListIn()
        );
    }

    /** Update an auth token's properties */
    public AdminAuthTokenUpdateOut update(
        final AdminAuthTokenUpdateIn adminAuthTokenUpdateIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.admin.auth-token.update",
            null,
            adminAuthTokenUpdateIn,
            AdminAuthTokenUpdateOut.class
        );
    }

    /** Return the role of the currently authenticated token */
    public AdminAuthTokenWhoamiOut whoami(
        final AdminAuthTokenWhoamiIn adminAuthTokenWhoamiIn
    ) throws DiomException {

        return this.client.executeRequest(
            "POST",
            "/api/v1.admin.auth-token.whoami",
            null,
            adminAuthTokenWhoamiIn,
            AdminAuthTokenWhoamiOut.class
        );
    }

    /** Return the role of the currently authenticated token */
    public AdminAuthTokenWhoamiOut whoami(
        
    ) throws DiomException {
        return this.whoami(
            new AdminAuthTokenWhoamiIn()
        );
    }
}

File: z-clients/python/diom/apis/admin_auth_token.py
# This file is @generated

from ..internal.api_common import ApiBase, parse_response
from ..models import (
    AdminAuthTokenCreateIn,
    AdminAuthTokenCreateOut,
    AdminAuthTokenDeleteIn,
    AdminAuthTokenDeleteOut,
    AdminAuthTokenExpireIn,
    AdminAuthTokenExpireOut,
    AdminAuthTokenListIn,
    AdminAuthTokenRotateIn,
    AdminAuthTokenRotateOut,
    AdminAuthTokenUpdateIn,
    AdminAuthTokenUpdateOut,
    AdminAuthTokenWhoamiIn,
    AdminAuthTokenWhoamiOut,
    ListResponseAdminAuthTokenOut,
)


class AdminAuthTokenAsync(ApiBase):
    async def create(
        self,
        admin_auth_token_create_in: AdminAuthTokenCreateIn,
    ) -> AdminAuthTokenCreateOut:
        """Create an auth token"""
        body = admin_auth_token_create_in.model_dump(exclude_none=True)

        response = await self._request_asyncio(
            method="post",
            path="/api/v1.admin.auth-token.create",
            body=body,
        )
        return parse_response(response, AdminAuthTokenCreateOut)

    async def expire(
        self,
        admin_auth_token_expire_in: AdminAuthTokenExpireIn,
    ) -> AdminAuthTokenExpireOut:
        """Expire an auth token"""
        body = admin_auth_token_expire_in.model_dump(exclude_none=True)

        response = await self._request_asyncio(
            method="post",
            path="/api/v1.admin.auth-token.expire",
            body=body,
        )
        return parse_response(response, AdminAuthTokenExpireOut)

    async def rotate(
        self,
        admin_auth_token_rotate_in: AdminAuthTokenRotateIn,
    ) -> AdminAuthTokenRotateOut:
        """Rotate an auth token, invalidating the old one and issuing a new secret"""
        body = admin_auth_token_rotate_in.model_dump(exclude_none=True)

        response = await self._request_asyncio(
            method="post",
            path="/api/v1.admin.auth-token.rotate",
            body=body,
        )
        return parse_response(response, AdminAuthTokenRotateOut)

    async def delete(
        self,
        admin_auth_token_delete_in: AdminAuthTokenDeleteIn,
    ) -> AdminAuthTokenDeleteOut:
        """Delete an auth token"""
        body = admin_auth_token_delete_in.model_dump(exclude_none=True)

        response = await self._request_asyncio(
            method="post",
            path="/api/v1.admin.auth-token.delete",
            body=body,
        )
        return parse_response(response, AdminAuthTokenDeleteOut)

    async def list(
        self,
        admin_auth_token_list_in: AdminAuthTokenListIn = AdminAuthTokenListIn(),
    ) -> ListResponseAdminAuthTokenOut:
        """List auth tokens for a given owner"""
        body = admin_auth_token_list_in.model_dump(exclude_none=True)

        response = await self._request_asyncio(
            method="post",
            path="/api/v1.admin.auth-token.list",
            body=body,
        )
        return parse_response(response, ListResponseAdminAuthTokenOut)

    async def update(
        self,
        admin_auth_token_update_in: AdminAuthTokenUpdateIn,
    ) -> AdminAuthTokenUpdateOut:
        """Update an auth token's properties"""
        body = admin_auth_token_update_in.model_dump(exclude_none=True)

        response = await self._request_asyncio(
            method="post",
            path="/api/v1.admin.auth-token.update",
            body=body,
        )
        return parse_response(response, AdminAuthTokenUpdateOut)

    async def whoami(
        self,
        admin_auth_token_whoami_in: AdminAuthTokenWhoamiIn = AdminAuthTokenWhoamiIn(),
    ) -> AdminAuthTokenWhoamiOut:
        """Return the role of the currently authenticated token"""
        body = admin_auth_token_whoami_in.model_dump(exclude_none=True)

        response = await self._request_asyncio(
            method="post",
            path="/api/v1.admin.auth-token.whoami",
            body=body,
        )
        return parse_response(response, AdminAuthTokenWhoamiOut)


class AdminAuthToken(ApiBase):
    def create(
        self,
        admin_auth_token_create_in: AdminAuthTokenCreateIn,
    ) -> AdminAuthTokenCreateOut:
        """Create an auth token"""
        body = admin_auth_token_create_in.model_dump(exclude_none=True)

        response = self._request_sync(
            method="post",
            path="/api/v1.admin.auth-token.create",
            body=body,
        )
        return parse_response(response, AdminAuthTokenCreateOut)

    def expire(
        self,
        admin_auth_token_expire_in: AdminAuthTokenExpireIn,
    ) -> AdminAuthTokenExpireOut:
        """Expire an auth token"""
        body = admin_auth_token_expire_in.model_dump(exclude_none=True)

        response = self._request_sync(
            method="post",
            path="/api/v1.admin.auth-token.expire",
            body=body,
        )
        return parse_response(response, AdminAuthTokenExpireOut)

    def rotate(
        self,
        admin_auth_token_rotate_in: AdminAuthTokenRotateIn,
    ) -> AdminAuthTokenRotateOut:
        """Rotate an auth token, invalidating the old one and issuing a new secret"""
        body = admin_auth_token_rotate_in.model_dump(exclude_none=True)

        response = self._request_sync(
            method="post",
            path="/api/v1.admin.auth-token.rotate",
            body=body,
        )
        return parse_response(response, AdminAuthTokenRotateOut)

    def delete(
        self,
        admin_auth_token_delete_in: AdminAuthTokenDeleteIn,
    ) -> AdminAuthTokenDeleteOut:
        """Delete an auth token"""
        body = admin_auth_token_delete_in.model_dump(exclude_none=True)

        response = self._request_sync(
            method="post",
            path="/api/v1.admin.auth-token.delete",
            body=body,
        )
        return parse_response(response, AdminAuthTokenDeleteOut)

    def list(
        self,
        admin_auth_token_list_in: AdminAuthTokenListIn = AdminAuthTokenListIn(),
    ) -> ListResponseAdminAuthTokenOut:
        """List auth tokens for a given owner"""
        body = admin_auth_token_list_in.model_dump(exclude_none=True)

        response = self._request_sync(
            method="post",
            path="/api/v1.admin.auth-token.list",
            body=body,
        )
        return parse_response(response, ListResponseAdminAuthTokenOut)

    def update(
        self,
        admin_auth_token_update_in: AdminAuthTokenUpdateIn,
    ) -> AdminAuthTokenUpdateOut:
        """Update an auth token's properties"""
        body = admin_auth_token_update_in.model_dump(exclude_none=True)

        response = self._request_sync(
            method="post",
            path="/api/v1.admin.auth-token.update",
            body=body,
        )
        return parse_response(response, AdminAuthTokenUpdateOut)

    def whoami(
        self,
        admin_auth_token_whoami_in: AdminAuthTokenWhoamiIn = AdminAuthTokenWhoamiIn(),
    ) -> AdminAuthTokenWhoamiOut:
        """Return the role of the currently authenticated token"""
        body = admin_auth_token_whoami_in.model_dump(exclude_none=True)

        response = self._request_sync(
            method="post",
            path="/api/v1.admin.auth-token.whoami",
            body=body,
        )
        return parse_response(response, AdminAuthTokenWhoamiOut)
